src/reader.rs

```diff
@@ -337,7 +337,9 @@ impl<'a> Reader<'a> {
             return Err(Error::UnsupportedCompVersion);
         }
 
-        if header.total_size as usize != blob.len() {
+        if header.total_size != blob.len() as u32
+            || header.total_size as usize != blob.len()
```

Comment:
Please explain why do we need two checks instead of a one?

Reply:
Because `as usize` may be a lossy conversion, then the might match just by accident. This effectively a full round trip an ensure that all the sizes compare correctly on all pointer sizes. Note that `as u_` can never lead to a result *mathematically* greater than the operand. Thus `num_u32 as usize as u32` asserts that `num_u32` fits a `usize`.

An alternative (but less clear imho) formulation would have been
```
header.total_size as usize as u32 != header.total_size
  || header.total_size as usize !=  blob.len()
```

Reply:
Oh, you probably mean architectures other than 32/64 bits.

Reply:
I would have used `TryFrom` but didn't want to introduce any new imports that would have required a compiler version bump.